### PR TITLE
Added a spec for new cloud_defend integration input

### PR DESCRIPTION
**Pull request by mitodrummer**
## What does this PR do?

Creates the spec for a new executable being added for container drift prevention.

## Why is it important?

A new cloud defend integration is being created by the AWP team to allow users to protect their containers.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

TODO

## Related issues
- Relates https://github.com/elastic/integrations/pull/4680

### CODE REVIEW

**Review by lrishi — APPROVED** [review]
Thanks Karl 🥇
Cc: @norrietaylor @m-sample

**Review by oren-zohar — COMMENTED** [review]
I think you also need to add the stream like other beats do here:
https://github.com/elastic/elastic-agent/blob/67021cfc1b66136e42a1570c559f5da4f11f68fe/specs/filebeat.spec.yml#L34

**Comment by oren-zohar** [comment]
I would also suggest following the arguments other beats pass, see here
https://github.com/elastic/elastic-agent/blob/67021cfc1b66136e42a1570c559f5da4f11f68fe/specs/filebeat.spec.yml#L21
you can ignore `"filebeat.config.modules.enabled=false"` in your case.

**Comment by mitodrummer** [comment]
I would imagine the test is failing because we probably don't have a cloud-defend artifact available yet cc @lrishi

File: specs/cloud-defend.spec.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - name: cloud_defend
+    description: "Cloud defend for containers"
+    platforms: &platforms
+      - container/amd64
+      - container/arm64
+    outputs: &outputs
+      - elasticsearch
+    command:
+      args: &args
+        - "-E"
```
Comment (lrishi) [inline specs/cloud-defend.spec.yml:12]: oh these will cause our program to fail. We aren't a beat, we're more standalone like endpoint.

Comment (mitodrummer) [reply specs/cloud-defend.spec.yml:12]: I see. Ok, i'll remove.

File: changelog/fragments/1669049916-CHANGELOG.asciidoc.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: feature
+
+# Change summary; a 80ish characters long description of the change.
+summary: Adds support for a new beat for container drift prevention.
```
Comment (cmacknz) [inline changelog/fragments/1669049916-CHANGELOG.asciidoc.yaml:14]: The changes here are enough to let the agent run your new input when it is manually packaged with the agent, but it won't be usable as part of any release until you get it hooked into to the agent release process. 

We may want to hold off on adding release notes until that is done, but if you plan to hook into the release process for 8.7 we can just leave this here.

File: internal/pkg/agent/application/monitoring/v1_monitor.go [hunk 3]
```diff
@@ -54,8 +54,8 @@ const (
 )
 
 var (
-	supportedComponents      = []string{"filebeat", "metricbeat", "apm-server", "auditbeat", "cloudbeat", "endpoint-security", "fleet-server", "heartbeat", "osquerybeat", "packetbeat"}
-	supportedBeatsComponents = []string{"filebeat", "metricbeat", "apm-server", "auditbeat", "cloudbeat", "heartbeat", "osquerybeat", "packetbeat"}
+	supportedComponents      = []string{"filebeat", "metricbeat", "apm-server", "auditbeat", "cloudbeat", "cloud-defend", "endpoint-security", "fleet-server", "heartbeat", "osquerybeat", "packetbeat"}
+	supportedBeatsComponents = []string{"filebeat", "metricbeat", "apm-server", "auditbeat", "cloudbeat", "cloud-defend", "heartbeat", "osquerybeat", "packetbeat"}
```
Comment (cmacknz) [inline internal/pkg/agent/application/monitoring/v1_monitor.go:58]:
```suggestion
	supportedBeatsComponents = []string{"filebeat", "metricbeat", "apm-server", "auditbeat", "cloudbeat", "heartbeat", "osquerybeat", "packetbeat"}
```

Is cloud-defend a Beat? If not you only want to be in the `supportedComponents` list.

Comment (mitodrummer) [reply internal/pkg/agent/application/monitoring/v1_monitor.go:58]: Not a beat. Thanks for the heads up.

File: changelog/fragments/1669049916-CHANGELOG.asciidoc.yaml (hunk 2 above)
Comment (mitodrummer) [reply changelog/fragments/1669049916-CHANGELOG.asciidoc.yaml:14]: cc @lrishi

Comment (lrishi) [reply changelog/fragments/1669049916-CHANGELOG.asciidoc.yaml]: Thanks, created a work item: https://github.com/elastic/anduril/issues/35

**Review by cmacknz — APPROVED** [review]

**Review by cmacknz — APPROVED** [review]

**Pull request merged**